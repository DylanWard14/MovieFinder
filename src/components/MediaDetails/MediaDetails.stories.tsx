import React from 'react';
import { withKnobs, text, number } from '@storybook/addon-knobs';
import styled from 'styled-components/macro';

import MediaDetails from './MediaDetails';

const Wrapper = styled.div`
    width: 100%;
    height: 100%;
    background: black;
`;

export default {
    title: 'Components/MediaDetails',
    decorators: [withKnobs],
};

export const DefaultExample = () => (
    <Wrapper>
        <MediaDetails
            title="Star Wars"
            tagLine="A long time ago in a galaxy far, far away..."
            overview="Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire."
            genres={[{ name: 'test' }]}
            productionCompanies={[{ name: 'test' }]}
            release="1977-05-25"
            runTime={121}
            boxOffice={7752345}
            averageVote="8.2 / 10"
        />
    </Wrapper>
);

export const PropsExample = () => (
    <Wrapper>
        <MediaDetails
            title={text('Title', 'Movie Title')}
            tagLine={text('Tagline', 'Movie Tagline')}
            overview={text('Overview', 'Movie Overview')}
            genres={[{ name: 'test' }]}
            productionCompanies={[{ name: 'test' }]}
            release={text('Release', 'Movie Release Date')}
            runTime={number('Runtime', 123)}
            boxOffice={number('Box Office', 5423667)}
            averageVote={text('Average Vote', 'Movie Average Vote')}
        />
    </Wrapper>
);
