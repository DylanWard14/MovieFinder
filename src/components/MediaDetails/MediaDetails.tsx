import React from 'react';
import styled from 'styled-components/macro';

import TextWithLabel from '../TextWithLabel/TextWithLabel';
import formatNumberAsThousands from '../../helpers/formatNumberAsThousands';

const Component = styled.div`
    padding: 1rem 1.5rem;
`;

const Title = styled.h1`
    text-transform: uppercase;
    ${props => props.theme.typography.heading};
    color: ${props => props.theme.colors.secondary};
`;

const TagLine = styled.p`
    ${props => props.theme.typography.largeLabel}
`;

const Overview = styled.p`
    ${props => props.theme.typography.paragraph}
    line-height: 1.5rem;
    margin-top: 0.5rem;
    margin-bottom: 2rem;
`;

const Stats = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding-right: 4rem;
`;

const StyledTextWithLabel = styled(TextWithLabel)`
    width: 10rem;
`;

interface MediaDetailsProps {
    title: string;
    tagLine: string;
    overview: string;
    genres?: { name: string }[];
    productionCompanies?: { name: string }[];
    release: string;
    runTime: number;
    boxOffice: number;
    averageVote: string;
}

const MediaDetails: React.FunctionComponent<MediaDetailsProps> = ({
    title,
    tagLine,
    overview,
    genres = [],
    productionCompanies = [],
    release,
    runTime,
    boxOffice,
    averageVote,
}) => {
    return (
        <Component>
            <Title id="movie-title">{title}</Title>
            <TagLine>{tagLine}</TagLine>
            <Overview>{overview}</Overview>
            <TextWithLabel
                label={genres?.reduce<string>((acc, item) => {
                    if (acc === '') {
                        return item.name;
                    }
                    return `${acc}, ${item.name}`;
                }, '')}
                content={productionCompanies.reduce<string>((acc, item) => {
                    if (acc === '') {
                        return item.name;
                    }
                    return `${acc}, ${item.name}`;
                }, '')}
            />
            <Stats>
                <StyledTextWithLabel label="Orignal Release:" content={release} labelSize="small" contentSize="large" />
                <StyledTextWithLabel
                    label="Running Time:"
                    content={`${runTime} mins`}
                    labelSize="small"
                    contentSize="large"
                />
            </Stats>
            <Stats>
                <StyledTextWithLabel
                    label="Box Office:"
                    content={`$${formatNumberAsThousands(boxOffice)}`}
                    labelSize="small"
                    contentSize="large"
                />
                <StyledTextWithLabel
                    label="Vote Average:"
                    content={`${averageVote} / 10`}
                    labelSize="small"
                    contentSize="large"
                />
            </Stats>
        </Component>
    );
};

export default MediaDetails;
