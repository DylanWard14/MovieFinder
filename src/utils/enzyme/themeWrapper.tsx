import React from 'react';
import { render, shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import myTheme from '../../theme';

export const renderWrapper = (component: React.ReactNode) => {
    return render(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const shallowWrapper = (component: React.ReactNode) => {
    return shallow(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const mountWrapper = (component: React.ReactNode) => {
    return mount(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};
