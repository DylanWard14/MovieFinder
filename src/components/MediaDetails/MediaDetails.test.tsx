import React from 'react';
import { render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import myTheme from '../../theme';
import MediaDetails from './MediaDetails';

describe('TextWithLabel', () => {
    it('renders with a label and content correctly', () => {
        const wrapper = render(
            <ThemeProvider theme={myTheme}>
                <MediaDetails
                    title="Star Wars"
                    tagLine="A long time ago in a galaxy far, far away..."
                    overview="Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire."
                    genres={[{ name: 'genre1' }, { name: 'genre2' }]}
                    productionCompanies={[{ name: 'company1' }, { name: 'company2' }]}
                    release="1977-05-25"
                    runTime={121}
                    boxOffice={775398007}
                    averageVote="8.2"
                />
            </ThemeProvider>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
