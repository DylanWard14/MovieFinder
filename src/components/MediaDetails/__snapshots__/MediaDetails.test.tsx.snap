// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWithLabel renders with a label and content correctly 1`] = `
<div
  class="MediaDetails__Component-h4ei75-0 kCELJj"
>
  <h1
    class="MediaDetails__Title-h4ei75-1 ksMIXu"
  >
    Star Wars
  </h1>
  <p
    class="MediaDetails__TagLine-h4ei75-2 dFiTEg"
  >
    A long time ago in a galaxy far, far away...
  </p>
  <p
    class="MediaDetails__Overview-h4ei75-3 gJIbCM"
  >
    Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire.
  </p>
  <div
    class="TextWithLabel__Component-sc-13uehc1-0 juyjpy"
  >
    <p
      class="TextWithLabel__Heading-sc-13uehc1-1 bTEzLT"
    >
      genre1, genre2
    </p>
    <p
      class="TextWithLabel__Content-sc-13uehc1-2 djIfsi"
    >
      company1, company2
    </p>
  </div>
  <div
    class="MediaDetails__Stats-h4ei75-4 isbobJ"
  >
    <div
      class="TextWithLabel__Component-sc-13uehc1-0 juyjpy MediaDetails__StyledTextWithLabel-h4ei75-5 dsWNqY"
    >
      <p
        class="TextWithLabel__Heading-sc-13uehc1-1 dvvKXU"
      >
        Orignal Release:
      </p>
      <p
        class="TextWithLabel__Content-sc-13uehc1-2 ZPJpN"
      >
        1977-05-25
      </p>
    </div>
    <div
      class="TextWithLabel__Component-sc-13uehc1-0 juyjpy MediaDetails__StyledTextWithLabel-h4ei75-5 dsWNqY"
    >
      <p
        class="TextWithLabel__Heading-sc-13uehc1-1 dvvKXU"
      >
        Running Time:
      </p>
      <p
        class="TextWithLabel__Content-sc-13uehc1-2 ZPJpN"
      >
        121 mins
      </p>
    </div>
  </div>
  <div
    class="MediaDetails__Stats-h4ei75-4 isbobJ"
  >
    <div
      class="TextWithLabel__Component-sc-13uehc1-0 juyjpy MediaDetails__StyledTextWithLabel-h4ei75-5 dsWNqY"
    >
      <p
        class="TextWithLabel__Heading-sc-13uehc1-1 dvvKXU"
      >
        Box Office:
      </p>
      <p
        class="TextWithLabel__Content-sc-13uehc1-2 ZPJpN"
      >
        $775,398,007
      </p>
    </div>
    <div
      class="TextWithLabel__Component-sc-13uehc1-0 juyjpy MediaDetails__StyledTextWithLabel-h4ei75-5 dsWNqY"
    >
      <p
        class="TextWithLabel__Heading-sc-13uehc1-1 dvvKXU"
      >
        Vote Average:
      </p>
      <p
        class="TextWithLabel__Content-sc-13uehc1-2 ZPJpN"
      >
        8.2 / 10
      </p>
    </div>
  </div>
</div>
`;
